### PR TITLE
Minor refactorings and code style changes

### PR DESCRIPTION
**Pull request by npathai**
- Removed several use of raw types
- Removed unnecessary throws clauses 
- Used lambda expressions wherever applicable 
- Used apt assertion methods for readability 
- Use of try with resources wherever applicable 
- Corrected incorrect order of assertXXX arguments

### CODE REVIEW

**Comment by iluwatar** [comment]
Looks good to me

File: data-mapper/src/test/java/com/iluwatar/datamapper/DataMapperTest.java [hunk 1]
```diff
@@ -58,12 +59,12 @@ public void testFirstDataMapper() {
     mapper.update(student);
 
     /* Check if student is updated in db */
-    assertEquals(mapper.find(student.getStudentId()).get().getName(), "AdamUpdated");
+    assertEquals("AdamUpdated", mapper.find(student.getStudentId()).get().getName());
```
[the remark this answers is not in the document]
Comment (npathai) [reply data-mapper/src/test/java/com/iluwatar/datamapper/DataMapperTest.java]: Fair point. I din't use `student.getName()` exactly but rather used a local variable at both places.

**Comment by npathai** [comment]
@iluwatar Merging this PR as I got two :+1: s

**Comment by iluwatar** [comment]
Thanks @npathai, well done :+1:

**Comment by iluwatar** [comment]
@all-contributors please add @npathai for code ideas maintenance mentoring question review

**Comment by iluwatar** [comment]
@all-contributors please add @npathai for code ideas maintenance mentoring question review

**Comment by iluwatar** [comment]
#798

**Pull request merged**